File: C++/2021/1stOct_abhilashkpy.cpp

/***
-----------------------------------------------------------------------------------------------------------------------------------------
DESCRIPTION: THIS IS A PROGRAM Program to print the Fibonacci Series up to n while replacing prime numbers, all multiples of 5 by 0.
-----------------------------------------------------------------------------------------------------------------------------------------
***/

#include <iostream>

using namespace std;

int main()
{
    int n, sum, a = 1, b = 1, marker;
    cout<<"Enter the limit:";
    cin>>n;
    cout<<a; 
    cout<<",";
    cout<<b;
    cout<<",";
    for (int i = 2; i < n; i++) {
    marker = 0;
    sum = a+b;
    if (sum %5 ==0)
        cout<<"0 ";
    else {
            for (int j = 2; j < sum /2; j++) 
                {
                 if (sum % j == 0)
                   {
                    cout<<sum<<" ";
                    marker = 1;
                    break;
                    }
                }
         if (marker == 0)
            cout<<"0 "; }
            a = b;
            b = sum;
        }
    

    return 0;
}

File: 1stOct_abhilashkpy.java
// PROGRAM TO: Print Fibonacci numbers up to 'n' numbers and replacing multiples of 5 and prime numbers with 0


import java.util.*;

public class fibonacci
{
    public static void main(String args[])
    {
        Scanner inp = new Scanner(System.in);

    int n, sum, n1 = 1, n2 = 1, marker;
    System.out.println("Enter the limit:");
    n = inp.nextInt();
    System.out.print(n1);
    System.out.print(" ");
    System.out.print(n2);
    System.out.print(" ");
    for (int i = 2; i < n; i++) {
    marker = 0;
    sum = n1+n2;
    if (sum %5 ==0)
        System.out.print("0 ");
    else {
            for (int j = 2; j < sum /2; j++) 
                {
                 if (sum % j == 0)
                   {
                    System.out.print(sum+" ");
                    marker = 1;
                    break;
                    }
                }
         if (marker == 0)
            System.out.print("0 "); 
        }
        
            n1 = n2;
            n2 = sum;
         }
    }
}
